File: core/feature46_test.go
package main

import "testing"

var (
	feature46Names      = []string{"ZMKS", "ZMGS", "L1", "L2", "M1", "M2"}
	feature46Accuracies = []string{"%.1f", "%.0f", "%.0f", "%.0f", "%.0f", "%.0f"}
	feature46Types      = []bool{true, true, false, false, false, false}
)

func Test01feature46(t *testing.T) {
	arguments := map[string]interface{}{"K": 200, "G": 300, "L": 200, "V": 200, "T": 6}
	expected := map[string]interface{}{"ZMKS": 104.7, "ZMGS": 157.1, "L1": []interface{}{112, 141, 283, 364}, "L2": []interface{}{112, 141, 320}, "M1": []interface{}{0, 135}, "M2": []interface{}{0}}
	genericTester(t, feature46(arguments), expected, feature46Names, feature46Accuracies, feature46Types)
}

File: core/feature46_test.js
var feature46Names = ["ZMKS", "ZMGS", "L1", "L2", "M1", "M2"];
var feature46Accuracies = [1, 0, 0, 0, 0, 0];
var feature46Types = [true, true, false, false, false, false];

function test01feature46() {
  var arguments = {"K": 200, "G": 300, "L": 200, "V": 200, "T": 6};
  var expected = {"ZMKS": 104.7, "ZMGS": 157.1, "L1": [112, 141, 283, 364], "L2": [112, 141, 320], "M1": [0, 135], "M2": [0]};
  genericTester(feature46_core(arguments), expected, feature46Names, feature46Accuracies, feature46Types);
}

test01feature46();
